system/Database/SQLite3/Forge.php

```diff
@@ -0,0 +1,226 @@
+     * @return    bool
+     */
+    public function createDatabase($db_name): bool
+    {
+        // In SQLite, a database is created when you connect to the database.
+        // We'll return TRUE so that an error isn't generated.
+        return TRUE;
+    }
+
+    //--------------------------------------------------------------------
+
+    /**
+     * Drop database
+     *
+     * @param    string $db_name
+     *
+     * @return    bool
+     * @throws \CodeIgniter\DatabaseException
+     */
+    public function dropDatabase($db_name): bool
+    {
+        // In SQLite, a database is dropped when we delete a file
+        if (!file_exists($db_name))
+        {
+            if ($this->db->DBDebug)
+            {
+                throw new DatabaseException('Unable to drop the specified database.');
+            }
+
+            return false;
+        }
+
+        // We need to close the pseudo-connection first
+        $this->db->close();
+        if ( ! @unlink($db_name))
+        {
+            if ($this->db->DBDebug)
+            {
+                throw new DatabaseException('Unable to drop the specified database.');
+            }
+
+            return false;
+        }
+
+        if ( ! empty($this->db->dataCache['db_names']))
+        {
+            $key = array_search(strtolower($db_name), array_map('strtolower', $this->db->dataCache['db_names']), true);
+            if ($key !== false)
+            {
+                unset($this->db->dataCache['db_names'][$key]);
+            }
+        }
+
+        return true;
+    }
+
+    //--------------------------------------------------------------------
+
+    /**
+     * ALTER TABLE
+     *
+     * @todo	implement drop_column(), modify_column()
+     * @param    string $alter_type ALTER type
+     * @param    string $table      Table name
+     * @param    mixed  $field      Column definition
+     *
+     * @return    string|array
+     */
+    protected function _alterTable($alter_type, $table, $field)
+    {
+        if (in_array($alter_type, ['DROP', 'CHANGE'], true))
+        {
+            return FALSE;
+        }
+        return parent::_alterTable($alter_type, $table, $field);
+    }
+
+    //--------------------------------------------------------------------
+
+    /**
+     * Process column
+     *
+     * @param	array	$field
+     * @return	string
+     */
+    protected function _processColumn($field)
+    {
+        return $this->db->escapeIdentifiers($field['name'])
+            . ' ' . $field['type']
+            . $field['auto_increment']
+            . $field['null']
+            . $field['unique']
+            . $field['default'];
+    }
+
+    //--------------------------------------------------------------------
+
+    /**
+     * Field attribute TYPE
+     *
+     * Performs a data type mapping between different databases.
+     *
+     * @param    array &$attributes
+     *
+     * @return    void
+     */
+    protected function _attributeType(&$attributes)
```

Comment:
If method names are prefixed with an underscore simply because they're a protected function, please remove the underscore. That's an old PHP4 habit we can finally get rid of. :) 

The only time you'll need to prefix is if the base class has an abstract method with an underscore.